modelscope/models/nlp/llama2/text_generation.py

```diff
@@ -186,3 +199,42 @@ def _reorder_cache(past_key_values, beam_idx):
                 past_state.index_select(0, beam_idx.to(past_state.device))
                 for past_state in layer_past), )
         return reordered_past
+
+    def chat(self, input: Dict, tokenizer) -> Dict:
+        import copy
+        gen_kwargs = copy.copy(input)
+        if 'utterance' not in input:
+            utterance: str = input['text']
+            gen_kwargs.pop('text')
+        else:
+            utterance: str = input['utterance']
+            gen_kwargs.pop('utterance')
+
+        if 'system' not in input:
+            system: str = ''
+        else:
+            system: str = input['system']
+            gen_kwargs.pop('system')
+
+        if 'history' not in input:
+            history = []
+        else:
+            history: List[Tuple] = copy.copy(input['history'])
+            gen_kwargs.pop('history')
+
+        if 'max_length' not in gen_kwargs:
+            gen_kwargs['max_length'] = 2048
+
+        prompt = get_chat_prompt(
+            system=system, utterance=utterance, history=history)
+        inputs = tokenizer(prompt, return_tensors='pt')
+        input_ids = inputs.input_ids.to(self.device)
+        generate_ids = self.generate(input_ids, **gen_kwargs)
+        response = tokenizer.batch_decode(
+            generate_ids,
+            skip_special_tokens=True,
+            clean_up_tokenization_spaces=False)[0]
+        response = response[len(prompt):].strip()
+        history.append((utterance, response))
```

Comment:
add history support in pipeline

Reply:
process it in the model for now, and extend it in the pipeline later